src/conduits/RwaMultiSwapOutputConduit.sol

```diff
@@ -0,0 +1,454 @@
+     * @param wad DAI amount.
+     */
+    function push(uint256 wad) external onlyMate {
+        _doPush(wad);
+    }
+
+    /**
+     * @notice Flushes out any DAI balance to `quitTo` address.
+     * @dev `msg.sender` must have received push access through `mate()`.
+     */
+    function quit() external onlyMate {
+        _doQuit(dai.balanceOf(address(this)));
+    }
+
+    /**
+     * @notice Flushes out the specified amount of DAI to the `quitTo` address.
+     * @dev `msg.sender` must have received push access through `mate()`.
+     * @param wad DAI amount.
+     */
+    function quit(uint256 wad) external onlyMate {
+        _doQuit(wad);
+    }
+
+    /**
+     * @notice Flushes out `amt` of `token` sitting in this contract to `usr` address.
+     * @dev Can only be called by the admin.
+     * @param token Token address.
+     * @param usr Destination address.
+     * @param amt Token amount.
+     */
+    function yank(
+        address token,
+        address usr,
+        uint256 amt
+    ) external auth {
+        GemAbstract(token).transfer(usr, amt);
+        emit Yank(token, usr, amt);
+    }
+
+    /**
+     * @notice Return Gem address of the selected PSM
+     * @return gem Gem address.
+     */
+    function gem() public view returns (address) {
+        return psm != address(0) ? GemJoinAbstract(PsmAbstract(psm).gemJoin()).gem() : address(0);
+    }
+
+    /**
+     * @notice Calculates the amount of GEM received for swapping `wad` of DAI.
+     * @param wad DAI amount.
+     * @return amt Expected GEM amount.
+     */
+    function expectedGemAmt(uint256 wad) public view returns (uint256 amt) {
+        require(psm != address(0), "RwaMultiSwapOutputConduit/psm-not-hooked");
+
+        uint256 to18ConversionFactor = 10**_sub(18, GemAbstract(gem()).decimals());
+        return _mul(wad, WAD) / _mul(_add(WAD, PsmAbstract(psm).tout()), to18ConversionFactor);
+    }
+
+    /**
+     * @notice Calculates the required amount of DAI to get `amt` amount of GEM.
+     * @param amt GEM amount.
+     * @return wad Required DAI amount.
+     */
+    function requiredDaiWad(uint256 amt) external view returns (uint256 wad) {
+        require(psm != address(0), "RwaMultiSwapOutputConduit/psm-not-hooked");
+
+        uint256 amt18 = _mul(amt, 10**_sub(18, GemAbstract(gem()).decimals()));
+        uint256 fee = _mul(amt18, PsmAbstract(psm).tout()) / WAD;
+        return _add(amt18, fee);
+    }
+
+    /**
+     * @notice Swaps the specified amount of DAI into GEM through the PSM and push it to the recipient address.
+     * @param wad DAI amount.
+     */
+    function _doPush(uint256 wad) internal {
+        require(to != address(0), "RwaMultiSwapOutputConduit/to-not-picked");
+        require(psm != address(0), "RwaMultiSwapOutputConduit/psm-not-hooked");
+
+        // We might lose some dust here because of rounding errors. I.e.: USDC has 6 dec and DAI has 18.
+        uint256 gemAmt = expectedGemAmt(wad);
```

Comment:
I just noticed that `require(psm != address(0), "RwaMultiSwapOutputConduit/psm-not-hooked");` is duplicated in this function and also in `expectedGemAmt`.

We either remove the check here or inline the calculation from `expectedGemAmt`.

Reply:
```suggestion
        require(to != address(0), "RwaMultiSwapOutputConduit/to-not-picked");

        // We might lose some dust here because of rounding errors. I.e.: USDC has 6 dec and DAI has 18.
        uint256 gemAmt = expectedGemAmt(wad);
```

Reply:
those methods are public and can be used by user to get the amounts needed. Not sure that we can inline this

Reply:
I meant we could copy the entire logic over instead of calling the function.
Or we drop the redundant check.